lib/nosedrum/component_handler.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Nosedrum.ComponentHandler do
+  @moduledoc """
+  Handles incommming interactions triggered by interacting with a component.
+
+  ## Register components
+  The component handler allows you to register a set of component ids to be
+  handled by a module that implements the `Nosedrum.ComponentInteraction`-
+  behaviour or any pid with the `register_components/2`-callback. 
+
+  When a pid is given to the `register_components/2`-callback, it will receive 
+  a message of the form `{:message_component_interaction, t:Nostrum.Struct.Interaction.t()}`.
```

Comment:
When will it receive the message? We should probably specify that it will receive it once the the component interaction is received by the gateway. Or is it actually sent that directly when calling it?

Also, I think it would be preferable to move this message to a type, so the linking of the Interaction struct works properly. Not sure there's another way to do it. Or does this link to the type these days?

Reply:
It does not work this way. Thank you for pointing that out!